tests/backends/test_kafka.py

```diff
@@ -133,6 +144,70 @@ def test_auto_offset_reset_error(self) -> None:
                     consumer.poll(10.0)  # XXX: getting the subcription is slow
 
 
+def test_cooperative_rebalancing() -> None:
+    configuration = build_kafka_configuration(
+        {"bootstrap.servers": os.environ.get("DEFAULT_BROKERS", "localhost:9092")}
+    )
+
+    partitions_count = 2
+
+    group_id = uuid.uuid1().hex
+    producer = KafkaProducer(configuration)
+
+    consumer_a = KafkaConsumer(
+        {
+            **configuration,
+            "auto.offset.reset": "earliest",
+            "enable.auto.commit": False,
+            "enable.auto.offset.store": False,
+            "group.id": group_id,
+            "session.timeout.ms": 10000,
+        },
+        incremental_cooperative=True,
+    )
+    consumer_b = KafkaConsumer(
+        {
+            **configuration,
+            "auto.offset.reset": "earliest",
+            "enable.auto.commit": False,
+            "enable.auto.offset.store": False,
+            "group.id": group_id,
+            "session.timeout.ms": 10000,
+        },
+        incremental_cooperative=True,
+    )
+
+    with get_topic(configuration, partitions_count) as topic, closing(
+        producer
+    ), closing(consumer_a), closing(consumer_b):
+        for i in range(10):
+            for j in range(partitions_count):
+                producer.produce(
+                    Partition(topic, j),
+                    KafkaPayload(None, f"{j}-{i}".encode("utf8"), []),
+                )
+
+        consumer_a.subscribe([topic])
```

Comment:
Why not providing callbacks so that you can assert that only one partition is transferred ?

Reply:
I guess that could've worked too. Still this works well enough in checking the same thing and it's copied from the test of the eager consumer.

Reply:
Right but there is no way to tell whether all partitions were revoked and then reassigned or whether the incremental assignment actually works.

Reply:
Yeah this was not about test Kafka consumer internals, just that we end up with one partition assigned to each consumer and consuming on both works ok.